### PR TITLE
Various compilation niceties and warnings that cropped up during the merge of 1.2

### PR DESCRIPTION
**Pull request by mbrookhart**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy<=1.15.0,>=1.8.2', 'requests<2.19.0,>=2.18.4', 'graphviz<0.9.0,>=0.8.1'], 'zip_safe': False}
+    kwargs = {'install_requires': ['numpy<=1.15.0,>=1.8.2', 'requests<2.19.0,>=2.18.4', 'graphviz<0.9.0,>=0.8.1', 'psutil>=5.4.3'], 'zip_safe': False}
```
Comment (cconvey) [inline python/setup.py]: Thanks for fixing this.  I think Lam made a different fix as part of commit 1101c160d7d0612b59becb44e32b895adc877bca (docker/scripts/run-mx-ngraph-validation-test.sh).  Should this PR undo Lam's change?

File: Makefile [hunk 2]
```diff
@@ -85,7 +85,7 @@ include $(TPARTYDIR)/mshadow/make/mshadow.mk
 include $(DMLC_CORE)/make/dmlc.mk
 
 # all tge possible warning tread
-WARNFLAGS= -Wall -Wsign-compare
+WARNFLAGS= -Wall -Wsign-compare  -Wno-comment
```
Comment (cconvey) [inline Makefile]: TBH I wish we were fixing this in ngraph instead of here, but this is okay enough IMO.

**Comment by mbrookhart** [comment]
TESTNOW

**Comment by mbrookhart** [comment]
TESTNOW

**Comment by mbrookhart** [comment]
TESTNOW

**Comment by mbrookhart** [comment]
TESTNOW

**Pull request merged**